ui/packages/catalog-ui-search/src/main/webapp/component/metacard-interactions/metacard-interactions.action.handlers.tsx

```diff
@@ -0,0 +1,179 @@
+import { Geometry } from 'wkx'
+
+import { WithBackboneProps } from '../../react-component/container/backbone-container'
+
+import * as CqlUtils from '../../js/CQLUtils'
+
+import * as CustomElements from '../../js/CustomElements'
+
+import * as Query from '../../js/model/Query'
+import * as QueryConfirmationView from '../confirmation/query/confirmation.query.view'
+import * as LoadingView from '../loading/loading.view'
+
+import * as store from '../../js/store'
+import * as user from '../../component/singletons/user-instance'
+
+const wreqr = require('wreqr')
+
+export type Props = {
+  model: {} | any
+  el: {} | any
+  extensions: {} | any
+  handleShare?: () => void
+} & WithBackboneProps
+
+export type Result = {
+  get: (key: any) => any
+  isWorkspace: () => boolean
+  isResource: () => boolean
+  isRevision: () => boolean
+  isDeleted: () => boolean
+  isRemote: () => boolean
+}
+
+export type Model = {
+  map: (
+    result: Result | any
+  ) =>
+    | {
+        id?: any
+        title?: any
+      }
+    | {}
+  toJSON: () => any
+  first: () => any
+  forEach: (result: Result | any) => void
+  find: (result: Result | any) => boolean
+} & Array<any>
+
+export const withCloseDropdown = (
+  context: Props,
+  action: (context: Props) => void
+) => {
+  context.el.trigger(`closeDropdown.${CustomElements.getNamespace()}`)
+  action(context)
+}
+
+export const handleAdd = (context: Props) =>
+  context.el
+    .find('.interaction-add > *')
+    .mousedown()
+    .click()
+
+export const handleDownload = (context: Props) => {
+  const openValidUrl = (result: Result) => {
+    const downloadUrl = result
+      .get('metacard')
+      .get('properties')
+      .get('resource-download-url')
+    downloadUrl && window.open(downloadUrl)
+  }
+
+  context.model.forEach(openValidUrl)
+}
+
+export const getGeoLocations = (model: Model) =>
+  model.reduce((locationArray: Array<string>, result: Result) => {
+    const location = result
+      .get('metacard')
+      .get('properties')
+      .get('location')
+
+    if (location) {
+      const geometry = Geometry.parse(location)
+      locationArray.push(`(${CqlUtils.buildIntersectCQL(geometry)})`)
```

Comment:
✏️ Consider moving the function call outside the template string, especially since the template string is already nested inside a function call.